### ISSUE TITLE
Data accessor function / computed value race condition?

### ISSUE THREAD

**Issue by chrisfrancis27** [ractivejs/ractive#2132]
Think I've found a regression with using `data` functions to access `computed` properties in the edge version.

1. [Here's a fiddle](http://jsfiddle.net/meofcksh/1/) demonstrating the behaviour against 'edge'
2. [Here's the same example](http://jsfiddle.net/meofcksh/2/) against 'latest'

The second example does what I'd expect, whereas the first has some odd behaviour. Clicking the "switch" button should, in theory, just toggle which array (`data1` or `data2`) is passed to the template. This will either render out 10 items (from 0 - 10) or 9 items (from 11 - 19). However, the first example renders out 9 items (0 - 9), which is a mix of the new array length with the old array values.

To summarise: when directly accessing a computed value in a template after it has been updated, the template always renders the correct value. However, when passing it to a parameterized function on the `data` object, it seems like the data function is run **before** the computed value is re-calculated.

I'm not entirely sure I'm correct in my assumptions and I haven't had a tonne of time to investigate this. Currently I've got a graph-drawing application with wonky diagonal axis ticks due to some function invocations returning the previous value in a loop. It's not every time, but it is consistent and sort of periodic, which makes me suspicious it may not be a race condition.

**Comment by Rich-Harris** [comment]
Thanks! Silly oversight on my part with #2122. Now fixed

**Comment by chrisfrancis27** [comment]
Well, that was quick! Cheers! :)
